dot/rpc/modules/state.go

```diff
@@ -245,10 +245,31 @@ func (sm *StateModule) GetPairs(_ *http.Request, req *StatePairRequest, res *Sta
 	return nil
 }
 
-// Call isn't implemented properly yet.
-func (sm *StateModule) Call(_ *http.Request, _ *StateCallRequest, _ *StateCallResponse) error {
-	_ = sm.networkAPI
-	_ = sm.storageAPI
+// Call makes a call to the runtime.
+func (sm *StateModule) Call(_ *http.Request, req *StateCallRequest, res *StateCallResponse) error {
+	var blockHash common.Hash
+	if req.Block == nil {
+		blockHash = sm.blockAPI.BestBlockHash()
+	} else {
+		blockHash = *req.Block
+	}
+
+	rt, err := sm.blockAPI.GetRuntime(blockHash)
+	if err != nil {
+		return err
+	}
+
+	request, err := common.HexToBytes(req.Params)
+	if err != nil {
+		return fmt.Errorf("cannot convert hex data %s to bytes: %w", req.Params, err)
+	}
+
+	response, err := rt.Exec(req.Method, request)
+	if err != nil {
+		return err
+	}
+
+	*res = StateCallResponse(common.BytesToHex(response))
```

Comment:
Should we check `res` is not a `nil` pointer? I guess it's unlikely, but checking it at the start of the method and returning a sentinel error `return fmt.Errorf("%w", ErrResponsePointerIsNil)` would be nice, if it's `nil`.

I also suppose we require this method signature, but if we don't, it would make more sense to return `StateCallResponse` (as value) instead of accepting a `*StateCallResponse` argument.

Reply:
I believe the module refactor is out of scope for this task. This is being worked on for `ParachainHost` runtime calls. Could you create an issue for this? I can check with @danforbes on how to prioritise this.